File: 소스 코드/C#/PPT Remote Viewer Server/Utils/Observers/Subjects/ScreenRenewalNotifier.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PPTRemoteViewerServer.Utils.Observers.Subjects
{
    public class ScreenRenewalNotifier : ScreenRenewalSubject
    {
        public override void Notify(object parameter)
        {
            Bitmap temp = null;

            if (parameter is Bitmap)
                temp = parameter as Bitmap;

             for (int i = 0; i < observers.Count; i++)
                 observers[i].OnScreenChanged(temp);
        }
    }
}

File: Source code/Android/PPT Remote Viewer/app/src/main/java/com/dku/dblab/android/pptRemoteViewer/utils/observers/subjects/ScreenRenewalNotifier.java
package com.dku.dblab.android.pptRemoteViewer.utils.observers.subjects;

import android.graphics.Bitmap;

public class ScreenRenewalNotifier extends ScreenRenewalSubject {

	@Override
	public void notify(Object parameter) {
		Bitmap temp = null;
		
		if (parameter instanceof Bitmap) {
			temp = (Bitmap)parameter;
		}
		
		for (int i = 0; i < observers.size(); i++) {
			observers.get(i).onScreenChanged(temp);
		}
	}

}
